src/utils.c

```diff
@@ -96,6 +98,87 @@ set_fd_nonblocking(int fd)
 	return true;
 }
 
+/*
+ * Verify string is in base64url encoding format
+ *
+ *\param s string to verify
+ *
+ *\return 0 on success, -1 on failure
+ */
+int
+verify_base64url_format(char *s)
+{
+	regex_t pattern;
+	int ret = -1;
+	char *regex = NULL;
+	char errbuf[256];
+
+	const char *alph_set = "[a-zA-Z0-9_\\-]";
+
+	if (!s)	{
+		return -1;
+	}
+
+	ret = asprintf(&regex, "^(%1$s{4})*(%1$s{4}|%1$s{3}=|%1$s{2}==)$",
```

Comment:
Let's hope we never need to touch this line! It's written in Malbolge right? :smile: